dotnet/src/Connectors/Connectors.Memory.Pinecone/PineconeVectorStoreRecordCollection.cs

```diff
@@ -0,0 +1,278 @@
+    private readonly ConcurrentDictionary<string, Sdk.Index<GrpcTransport>> _indexMap = new();
+
+    /// <inheritdoc />
+    public string CollectionName { get; }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PineconeVectorStoreRecordCollection{TRecord}"/> class.
+    /// </summary>
+    /// <param name="pineconeClient">Pinecone client that can be used to manage the collections and vectors in a Pinecone store.</param>
+    /// <param name="options">Optional configuration options for this class.</param>
+    /// <exception cref="ArgumentNullException">Thrown if the <paramref name="pineconeClient"/> is null.</exception>
+    /// <param name="collectionName">The name of the collection that this <see cref="PineconeVectorStoreRecordCollection{TRecord}"/> will access.</param>
+    /// <exception cref="ArgumentException">Thrown for any misconfigured options.</exception>
+    public PineconeVectorStoreRecordCollection(Sdk.PineconeClient pineconeClient, string collectionName, PineconeVectorStoreRecordCollectionOptions<TRecord>? options = null)
+    {
+        Verify.NotNull(pineconeClient);
+
+        this._pineconeClient = pineconeClient;
+        this.CollectionName = collectionName;
+        this._options = options ?? new PineconeVectorStoreRecordCollectionOptions<TRecord>();
+
+        if (this._options.MapperType == PineconeRecordMapperType.PineconeVectorCustomMapper)
+        {
+            if (this._options.VectorCustomMapper is null)
+            {
+                throw new ArgumentException($"The {nameof(PineconeVectorStoreRecordCollectionOptions<TRecord>.VectorCustomMapper)} option needs to be set if a {nameof(PineconeVectorStoreRecordCollectionOptions<TRecord>.MapperType)} of {nameof(PineconeRecordMapperType.PineconeVectorCustomMapper)} has been chosen.", nameof(options));
+            }
+
+            this._mapper = this._options.VectorCustomMapper;
+        }
+        else
+        {
+            this._mapper = new PineconeVectorStoreRecordMapper<TRecord>(new PineconeVectorStoreRecordMapperOptions
+            {
+                VectorStoreRecordDefinition = this._options.VectorStoreRecordDefinition
+            });
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task<bool> CollectionExistsAsync(CancellationToken cancellationToken = default)
+    {
+        var result = await this.RunOperationAsync(
+            CollectionExistsName,
+            async () =>
+            {
+                var collections = await this._pineconeClient.ListIndexes(cancellationToken).ConfigureAwait(false);
+
+                return collections.Any(x => x.Name == this.CollectionName);
+            }).ConfigureAwait(false);
+
+        return result;
+    }
+
+    /// <inheritdoc />
+    public Task CreateCollectionAsync(CancellationToken cancellationToken = default)
+    {
+        throw new KernelException($"Index creation is not supported within the {nameof(PineconeVectorStoreRecordCollection<TRecord>)}. " +
```

Comment:
context: https://github.com/microsoft/semantic-kernel/pull/770/files#r1191722828

Reply:
Can we add a code comment here to explain that index creation can take a long time, which is why we are not supporting at at this time?

Reply:
added support for collection creation, defaulting to serverless as a reasonable default